views/Pokedex/Boomsweeper/Boomsweeper.jsx

```diff
@@ -0,0 +1,667 @@
+      break;
+
+    case ACTION.reset:
+      newItem.mineState = MINE_STATE.default;
+      newItem.dexState = DEX_STATE.none;
+      newItem.wasRevealedDirectly = false;
+      newItem.wasTriggeredFromExplosion = false;
+      break;
+  }
+
+  return clone;
+}
+
+function explode(grid, row, column) {
+  grid[row][column].mineState = MINE_STATE.exploded;
+  reveal(grid, row-1, column-1, true);
+  reveal(grid, row-1, column, true);
+  reveal(grid, row-1, column+1, true);
+  reveal(grid, row, column-1, true);
+  reveal(grid, row, column+1, true);
+  reveal(grid, row+1, column-1, true);
+  reveal(grid, row+1, column, true);
+  reveal(grid, row+1, column+1, true);
+}
+
+function reveal(grid, row, column, fromExplosion = false, direct = false) {
+  if (
+    row >= 0 &&
+    row < grid.length &&
+    column >= 0 &&
+    column < grid[row].length &&
+    grid[row][column].mineState !== MINE_STATE.revealed &&
+    grid[row][column].mineState !== MINE_STATE.exploded
+  ) {
+    grid[row][column].mineState = MINE_STATE.revealed
+
+    if (
+      (direct && grid[row][column].isMine) ||
+      grid[row][column].numberOfAdjacentMines === 0
+    ) {
+      revealAround(grid, row, column)
+    }
+
+    if (fromExplosion) {
+      grid[row][column].wasTriggeredFromExplosion = true
+    }
+
+    if (direct) {
+      grid[row][column].wasRevealedDirectly = true
+    }
+  }
+}
+
+function revealAround(grid, row, column) {
+  reveal(grid, row-1, column-1);
+  reveal(grid, row-1, column);
+  reveal(grid, row-1, column+1);
+  reveal(grid, row, column-1);
+  reveal(grid, row, column+1);
+  reveal(grid, row+1, column-1);
+  reveal(grid, row+1, column);
+  reveal(grid, row+1, column+1);
+}
+
+function numberOfItemsWhere(grid, condition) {
+  return grid.flat().reduce((number, item) => {
+    return number + (condition(item) ? 1 : 0)
+  }, 0);
+}
+
+export default function Boom({
+  seed,
+  layoutType,
+}) {
+  const [grid, setGrid] = useState([]);
+  const [currentClickAction, setCurrentClickAction] = useState();
+  const [searchTerm, setSearchTerm] = useState();
+  const [isShowingNewGridModal, setIsShowingNewGridModal] = useState();
+
+  useEffect(() => {
+    setGrid(generateGrid(
+      seed,
+      NATIONAL_DEX,
+      layoutType,
+      Math.ceil(Math.sqrt(NATIONAL_DEX.length)),
+      numberOfMines
+    ));
+
+    setCurrentClickAction(ACTION.mark_as_safe);
+    setSearchTerm('');
+    setIsShowingNewGridModal(false);
+  }, [seed, layoutType]);
+
+  const cellClickAction = (item) => {
+    setSearchTerm('');
+    setGrid(performActionOnItem(grid, item, currentClickAction));
+  };
+
+  const cellRightClickAction = (item) => {
+    setSearchTerm('');
+    setGrid(performActionOnItem(grid, item, ACTION.flag));
+  };
+
+  const pokemonLayoutName = (() => {
+    switch (layoutType) {
+      case LAYOUT_TYPE.numerical: return 'Numerical';
+      case LAYOUT_TYPE.alphabetical: return 'Alphabetical';
+      case LAYOUT_TYPE.random: return 'Randomized';
```

Comment:
Minor idea: `Numerical` should be the value for `LAYOUT_TYPE.numerical` so you don't need this helper function. Then, if you need it lowercased for something else, you can always do `LAYOUT_TYPE.numerical.toLowercase()`